docs/19_security_audit_document.md

```diff
@@ -0,0 +1,230 @@
+
+_Note: Metrics does not duplicate the user's authentication data to the third-party databases! All authentication-related data is managed by the Firebase Authentication and is passed to it using official packages._
+
+## Database
+
+Metrics Web Application uses the `Firebase Cloud Firestore` as a database.
+
+Consider the following [document that describes the data model used in `Firebase Cloud Firestore`](https://firebase.google.com/docs/firestore/data-model).
+
+The application uses the `Firebase Cloud Firestore Security Rules` to protect the data stored in the `Cloud Firestore`. The `Security Rules` describe conditions that should be met to access or modify the data these rules are protecting. If the CRUD request does not satisfy the appropriate set of conditions, this request is refused as having insufficient permissions.
+
+### The `allowed_email_domains` collection
+
+The `allowed_email_domains` collection contains the allowed email domains for the `Google Sign-In` method. The single document of this collection stands for the email domain that is allowed and may be used to authorize a user.
+
+The documents of this collection do not have any fields. Instead, the ID of the document represents a single email domain. This prevents duplicates of email domains and simplifies validating as documents aren't to be fetched.
+
+Here is a table of security rules applied to the `allowed_email_domains` collection.
+
+| Operation       | Security Rules |
+|-----------------|----------------|
+| `read`, `write` | `Prohibited`   |
+
+### The `build` collection
+
+The `build` collection contains build data for all projects within the Metrics Web Application. The single document of this collection stands for a single build for a specific project.
+
+Consider the following table that describes the fields of a document within the `build` collection:
+
+| Field        | Description                                                 |
+|--------------|-------------------------------------------------------------|
+|`buildNumber` | A number of this build.                                     |
+|`buildStatus` | A resulting status of this build.                           |
+|`coverage`    | A test coverage percent of this build.                      |
+|`duration`    | A duration of this build excluding the queue time.          |
+|`projectId`   | An id of the project this build belongs to.                 |
+|`startedAt`   | A date-time of this build's start.                          |
+|`url`         | A URL of the source control revision used to run the build. |
+|`workflowName`| A name of the workflow executed this build.                 |
+
+Here is a table of security rules applied to the `build` collection.
+
+| Operation          | Security Rules                  |
+|--------------------|---------------------------------|
+| `read`             | `Authorization`                 |
```

Comment:
Would it be great to explain each `security rule` a bit wider? I mean that the `Authorization` or `isBuildValid` may be incomprehensible without additional context. I think it would be nice to mention that, for example, `Authentication` means that the user is authenticated and has a valid email domain somewhere.